### PR TITLE
Add basic UsersUnsubscribe page

### LINKED ISSUE #7840: Add UsersUnsubscribe page
Supports #7656 

---

We want to add a new page for the unsubscribe feature. This issue is about creating a new route + a simple component for that. It should look similar to the old page:

Screen Shot 2019-04-05 at 15 24 45

It will NOT unsubscribe the user. It's a pure UI (and basic) component.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #7840

---

This patch adds a new `UsersUnsubscribe` page component. It won't be enabled in -dev because the route is redirected to addons-server at the nginx level. This patch supports #7656. It does NOT perform any action yet.

Being logged in or out does not matter because we'll use the hash/token in the URL. That's why the two screenshots thereafter are the same. The UI is similar to the old one but I did not add two links to edit the notification settings/profile. It's the same page, so I added a sentence that explains what a user can do after having landed to this page. That's what I'd expect when landing to this page. 

## Screenshots

User is logged in:

Screen Shot 2019-04-05 at 16 24 27

User is not logged in:

Screen Shot 2019-04-05 at 16 24 24

### CODE REVIEW

File: src/amo/pages/UsersUnsubscribe/index.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        <Card header={i18n.gettext('You are successfully unsubscribed!')}>
+          <p
+            className="UsersUnsubscribe-content-explanation"
+            // eslint-disable-next-line react/no-danger
+            dangerouslySetInnerHTML={sanitizeHTML(
+              i18n.sprintf(
+                i18n.gettext(
+                  `The email address %(strongStart)s%(email)s%(strongEnd)s will
+                  no longer get messages when:`,
```
Comment (kumar303) [inline src/amo/pages/UsersUnsubscribe/index.js]: A `translators:` comment explaining that a list of notifications will be displayed under this prompt would probably be helpful.

File: src/amo/utils/notifications.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      return i18n.gettext('someone writes a review of my add-on');
+    case 'reply':
+      return i18n.gettext('an add-on developer replies to my review');
+    case 'reviewer_reviewed':
+      return i18n.gettext('my add-on is reviewed by a reviewer');
+    case 'upgrade_fail':
+      return i18n.gettext(`my add-on's compatibility cannot be upgraded`);
+    case 'upgrade_success':
+      return i18n.gettext(`my add-on's compatibility is upgraded successfully`);
+    default:
```
Comment (kumar303) [inline src/amo/utils/notifications.js]: Is there any reason why `return null` can't go here? I guess Flow complained?

File: tests/unit/amo/components/TestRoutes.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ describe(__filename, () => {
   it('renders the routes for the amo app', () => {
     const root = render();
 
-    expect(root.find(Route)).toHaveLength(32);
+    expect(root.find(Route)).toHaveLength(33);
```
Comment (kumar303) [inline tests/unit/amo/components/TestRoutes.js]: Marie Kondo would not like this test 🗑

File: src/amo/utils/notifications.js (hunk 2 above)
Comment (willdurand) [reply src/amo/utils/notifications.js]: It works actually.

**Pull request merged**